python/iron/config.py

```diff
@@ -6,8 +6,56 @@
 #
 # (c) Copyright 2025 Advanced Micro Devices, Inc.
 
+import subprocess
+
+from .device import NPU1Col4, NPU2
+
+
+def detect_npu_device():
+    """Detects the current device in the system.
+       This assumes XRT and XDNA driver is installed
+       and the system has NPU hardware.
+
+    Returns:
+        The current system device.
+    """
+    try:
+        # Run `xrt-smi examine` and capture output
+        result = subprocess.run(
+            ["/opt/xilinx/xrt/bin/xrt-smi", "examine"],
+            check=True,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.DEVNULL,
+            text=True,
+        )
+        output = result.stdout
+
+        # Match strings for NPU2
+        # Set's generic "whole array" devices, this is overkill...
+        if any(
+            keyword.lower() in output.lower()
+            for keyword in [
+                "NPU Strix",
+                "NPU Strix Halo",
+                "NPU Kracken",
+                "RyzenAI-npu4",
+                "RyzenAI-npu6",
+            ]
+        ):
+            return NPU2()
+        else:
+            return NPU1Col4()
```

Comment:
It would be nice for future proofing to explicitly match Phoenix devices here and fail if it doesn't match a known string.